core/src/main/java/io/confluent/kafka/schemaregistry/kafka/SchemaRegistryCoordinator.java

```diff
@@ -0,0 +1,220 @@
+    } while (remaining > 0);
+  }
+
+  @Override
+  public List<JoinGroupRequest.ProtocolMetadata> metadata() {
+    ByteBuffer metadata = SchemaRegistryProtocol.serializeMetadata(identity);
+    return Collections.singletonList(
+        new JoinGroupRequest.ProtocolMetadata(SR_SUBPROTOCOL_V0, metadata)
+    );
+  }
+
+  @Override
+  protected void onJoinComplete(
+      int generation,
+      String memberId,
+      String protocol,
+      ByteBuffer memberAssignment
+  ) {
+    assignmentSnapshot = SchemaRegistryProtocol.deserializeAssignment(memberAssignment);
+    listener.onAssigned(assignmentSnapshot, generation);
+  }
+
+  @Override
+  protected Map<String, ByteBuffer> performAssignment(
+      String kafkaLeaderId, // Kafka group "leader" who does assignment, *not* the SR master
+      String protocol,
+      Map<String, ByteBuffer> allMemberMetadata
+  ) {
+    log.debug("Performing assignment");
+
+    Map<String, SchemaRegistryIdentity> memberConfigs = new HashMap<>();
+    for (Map.Entry<String, ByteBuffer> entry : allMemberMetadata.entrySet()) {
+      SchemaRegistryIdentity identity
+          = SchemaRegistryProtocol.deserializeMetadata(entry.getValue());
+      memberConfigs.put(entry.getKey(), identity);
+    }
+
+    Set<String> members = memberConfigs.keySet();
+
+    // Compute the leader as the master-eligible member with the "smallest" (lexicographically) ID.
+    // This doesn't guarantee a member will stay master until it leaves the group, but should
+    // usually keep the master assigned to the same member across rebalances.
+    SchemaRegistryIdentity masterIdentity = null;
+    String masterKafkaId = null;
+    Set<String> urls = new HashSet<>();
+    for (String kafkaMemberId : members) {
+      SchemaRegistryIdentity memberIdentity = memberConfigs.get(kafkaMemberId);
+      urls.add(memberIdentity.getUrl());
+      boolean eligible = memberIdentity.getMasterEligibility();
+      boolean smallerIdentity = masterIdentity == null
+                                || memberIdentity.getUrl().compareTo(masterIdentity.getUrl()) < 0;
+      if (eligible && smallerIdentity) {
+        masterKafkaId = kafkaMemberId;
+        masterIdentity = memberIdentity;
+      }
+    }
+    short error = SchemaRegistryProtocol.Assignment.NO_ERROR;
+    if (masterIdentity == null) {
+      error = SchemaRegistryProtocol.Assignment.NO_MASTER_ELIGIBLE_MEMBERS;
+    }
+
+    // Validate that schema registry instances aren't trying to use the same URL
+    if (urls.size() != members.size()) {
+      log.error("Found duplicate URLs for schema registry group members. This indicates a "
+                + "misconfiguration and is common when executing in containers. Use the host.name "
+                + "configuration to set each instance's advertised host name to a value that is "
+                + "routable from all other schema registry instances.");
+      error = SchemaRegistryProtocol.Assignment.DUPLICATE_URLS;
```

Comment:
not seeing anything being done with the error, unless i'm missing something. We would ideally have below behavior

If the master eligible node with a duplicate url is elected master then other  nodes should fail startup instead of just error